docs/reference/rtu-kernel-config-options.rst

```diff
@@ -0,0 +1,194 @@
+
+
+>>>>
+
+Basic preemption configurations
+-------------------------------
+Preemption refers to the ability of an operating system to interrupt the execution of a running task or process. 
+This interruption is done in order to give priority to another task.
+This feature is essential to reduce latency and improve the responsiveness of the system. 
+Real-time applications are the ones who most benefit from it, where deterministic response times are critical.
+
+ARCH_SUPPORTS_RT
+~~~~~~~~~~~~~~~~
+This is a boolean value that indicates if the architecture supports real-time features.
+
+PREEMPTION
+~~~~~~~~~~
+The primary config, that enables the preemptible kernel.
+This configuration needs to be enabled in order to enable the other preemption levels and configurations.
+
+PREEMPT_COUNT
+~~~~~~~~~~~~~
+Once the previous option get enabled, this configuration enables the counting of preemption.
+Needed for kernel APIs like `preempt_count()`_.
+
+PREEMPT_RT
+~~~~~~~~~~
+This configuration enables a fully preemptible kernel.
+There are `other levels of preemption`_ available in the kernel, but the PREEMPT_RT configuration is the most aggressive.
+The way that it works is by replacing some kernel primitives related to `locking`_ with preemptible priority-inheritance aware variants.
+This enables mechanisms that can break up long non-preemptible sections of code.
+However, this configuration doesn't apply for very low level and critical sections of the kernel, such as entry code, scheduler, and low level interrupt handlers.
+
+
+>>>>
+
+Kernel internals configurations
+-------------------------------
+Next configurations are related to kernel internals.
+
+PREEMPT_RCU
+~~~~~~~~~~~
+Enables preemption in the context of the synchronization mechanism `RCU - Read-copy update`_.
+This is done by selecting the RCU implementation designed for very large `Symmetric multiprocessing (SMP)`_ systems that usually had hundreds of CPUs and that requires real-time response times.
+However this also scales down nicely to smaller systems.
+
+RCU_BOOST
+~~~~~~~~~
+Working together with previous configuration, this one boosts the priority of preempted RCU readers that block the current preemptible RCU grace period for too long.
+This also prevents heavy loads from blocking RCU callback invocation. 
+
+RCU_NOCB_CPU
+~~~~~~~~~~~~
+Used to reduce OS jitter for aggressive HPC or real-time workloads. 
+It offloads RCU callback invocation to energy-efficient CPUs in asymmetric multiprocessors.
+The reduced jitter comes at the cost of increased `call_rcu()`_ overhead and potentially higher context-switch rates.
+This option creates kthreads ("``rcuox/N``") to invoke callbacks on specified CPUs, where "``N``" is the CPU being offloaded and "``x``" will be "``p``" for RCU-preempt (**PREEMPTION kernels**) and "``s``" for RCU-sched (**non PREEMPTION kernels**).
+Affinity or cgroups can be used to control the CPU set for the kthreads.
+
```

Comment:
Like Farshid said, it would be good to mention RCU_NOCB_CPU_CB_BOOST. Info can be found at https://cateee.net/lkddb/web-lkddb/RCU_NOCB_CPU_CB_BOOST.html and there are more details in the original patch thread: https://lore.kernel.org/lkml/20220511211059.GF1790663@paulmck-ThinkPad-P17-Gen-1/

Reply:
Thank you for the suggestion! Added!